### PR TITLE
LogFactory - Marked broken constructor as obsolete. It is a trap.

### PR DESCRIPTION
**Pull request by snakefoot**
See also: https://stackoverflow.com/questions/60803043/nlog-for-logging-using-nlog-api-into-application-insight-table-storage

### CODE REVIEW

File: src/NLog/Config/XmlLoggingConfiguration.cs [hunk 1]
```diff
@@ -196,11 +196,22 @@ internal XmlLoggingConfiguration([NotNull] string xmlContents, [CanBeNull] strin
         /// <summary>
         /// Parse XML string as NLog configuration
         /// </summary>
-        /// <param name="xml">NLog configuration</param>
+        /// <param name="xml">The XML contents.</param>
```
Comment (304NotModified) [inline src/NLog/Config/XmlLoggingConfiguration.cs:199]: proposal: "NLog configuration as XML string"

(I find The XML contents confusing)

Comment (304NotModified) [reply src/NLog/Config/XmlLoggingConfiguration.cs:199]: same below

File: src/NLog/LogFactory.cs [hunk 2]
```diff
@@ -114,6 +114,7 @@ public LogFactory()
         /// Initializes a new instance of the <see cref="LogFactory" /> class.
         /// </summary>
         /// <param name="config">The config.</param>
+        [Obsolete("Constructor with LoggingConfiguration as parameter should not be used. Instead provide LogFactory as parameter when constructing LoggingConfiguration.")]
```
Comment (304NotModified) [inline src/NLog/LogFactory.cs:117]: add the text, obsolete in NLog 5? (or something like that)

Comment (snakefoot) [reply src/NLog/LogFactory.cs]: Fixed

File: src/NLog/Config/XmlLoggingConfiguration.cs (hunk 1 above)
Comment (snakefoot) [reply src/NLog/Config/XmlLoggingConfiguration.cs]: Fixed

**Comment by 304NotModified** [comment]
Thanks!

**Pull request merged**